### ISSUE TITLE
[0.7] Wrong result with repacking

### ISSUE THREAD

**Issue by hajduakos** [SRI-CSL/solidity#161]
`structs/LocalStorageRepack.sol` reports an assertion failure which should hold.

```
test/solc-verify/structs/LocalStorageRepack.sol [ --show-warnings ]
1c1,2
< LocalStorageRepack::[fallback]: OK
---
> LocalStorageRepack::[receive]: ERROR
>  - test/solc-verify/structs/LocalStorageRepack.sol:36:9: Assertion might not hold.
3c4
< No errors found.
---
> Errors were found by the verifier.
```

**Comment by hajduakos** [comment]
Was tricky to find, but there was a mismatch between indexes due to replacing a loop. Fixed in 530e16a

**Comment by dddejan** [comment]
Ah sorry about that. I had a feeling that loop replacement was not a good idea but didn't go back to check.
